app/controllers/curation_concerns/generic_works_controller.rb

```diff
@@ -30,6 +31,13 @@ def identifiers
     end
   end
  
+  def assign_visibility
+    if (params["generic_work"]["visibility"]=="true")
```

Comment:
I suggest using a different parameter in the form.  Checking the value like this on visibility implies that 'true' is a valid value for visibility.  Perhaps `:is_draft?` or `draft_status` for the form parameter name.